pyqtgraph/functions.py

```diff
@@ -1185,7 +1193,7 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False, output=None
                     maxVal = xp.nextafter(maxVal, 2*maxVal)
                 rng = maxVal-minVal
                 rng = 1 if rng == 0 else rng
-                data = rescaleData(data, scale/rng, minVal, dtype=dtype)
+                data = rescaleData(data, scale/rng, minVal, dtype=dtype, log=log)
```

Comment:
I think these two changes will be insufficient to have `makeARGB` properly support log-scale image processing. Are the `lut` and `levels` args assumed to have already been adjusted to account for the log scale? I think we should omit changes to this method entirely, rather than implement them halfway and imply otherwise.

Reply:
Though this may be more sufficient than I understand. Providing an example use or test would help me. For now, though, if a user wants to process the image with log10, my first impulse would be to recommend they do so before sending it to `makeARGB`, and they could, at that time, adjust `lut` and `levels` to match their expectations.

Reply:
I am not that familiar with how the processing in ImageItem works, but my understanding is that moving the log10 inside rescaleData would also require a recalculation of the logarithm on each re-render, even if the image hasn't changed. I suspect that this would make interactive adjustments to the LUT slower than necessary.